File: 프로그래머스/lv0/120846. 합성수 찾기/합성수 찾기.cs
using System;
using System.Linq;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        
        answer = Enumerable.Range(1, n).Where(k => IsCompound(k)).Count();
        
        return answer;
    }
    
    public bool IsCompound(int num)
    {
        int count = 0;
        for(int i = 1; i <= num; i++)
        {
            if(num % i == 0)
            {
                count++;
                if(count >= 3)
                    return true;
            }
        }
        return false;
    }
}

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.cpp
#include <string>

using namespace std;

int solution(int n) {
    
    int* count = new int[n + 1]();
    for (int i = 2; i <= n; ++i)
    {
        for (int k = i; k <= n; k += i)
            count[k]++;
    }
    
    int answer = 0;
    for (int i = 1; i <= n; ++i)
    {
        if (count[i] >= 2) // 1은 모든수의 약수라 2개만 있으면 됨
            ++answer;
    }
    
    delete[] count;
    return answer;
}